src/build/arg/arg_value.rs

```diff
@@ -0,0 +1,71 @@
+use std::fmt::Display;
+
+/// TODO
+#[derive(Debug, Default, Clone)]
+pub struct ArgValue<'help> {
+    pub(crate) value: &'help str,
+    pub(crate) about: Option<&'help str>,
+    pub(crate) hidden: bool,
+}
+
+impl Display for ArgValue<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.value.contains(char::is_whitespace) {
+            write!(f, "{:?}", self.value)
+        } else {
+            write!(f, "{}", self.value)
+        }
+    }
+}
+
+impl<'help> From<&'help str> for ArgValue<'help> {
+    fn from(s: &'help str) -> Self {
+        Self::new(s)
+    }
+}
+
+/// Getters
+impl<'help> ArgValue<'help> {
+    /// TODO
+    #[inline]
+    pub fn get_value(&self) -> &str {
+        self.value
+    }
+
+    /// TODO
+    #[inline]
+    pub fn get_about(&self) -> Option<&str> {
+        self.about
+    }
+
+    /// TODO
+    #[inline]
+    pub fn get_hidden(&self) -> bool {
+        self.hidden
+    }
+}
+
+impl<'help> ArgValue<'help> {
+    /// TODO
+    pub fn new<S: Into<&'help str>>(v: S) -> Self {
```

Comment:
Please spell out `v` to `value` so people have a hint as to what this constructor is setting

Reply:
`value` or `name`? I had value first, but then changed to name to match the proposed implementation in #2756

Reply:
I was being sloppy with that proposal and not double checking things.   `value` seems more appropriate

Reply:
Not sure if name might be more consistent with other places after all:
```rs
#[derive(ArgEnum, Debug, PartialEq)]
enum GeneratorChoice {
    Bash,
    Elvish,
    Fish,
    #[clap(name = "powershell")]
    PowerShell,
    Zsh,
}
```

Reply:
@epage @pksunkara What do you think about the naming here, was just starting on the documentation and therefor wondering.

`name` and `alias` is used for everything like this in clap, as far as I can tell. 

And while it will be the value of the argument, it arguably is the name of the ArgValue.

Reply:
Fun side effects of reusing / being consistent in the derives :)

I can see switching to name.